src/storage_common.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) The ThingSet Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <thingset.h>
+#include <thingset/sdk.h>
+#include <thingset/storage.h>
+
+LOG_MODULE_REGISTER(thingset_storage_common, CONFIG_THINGSET_SDK_LOG_LEVEL);
+
+static struct k_work_delayable storage_work;
+
+static bool storage_save_inhibit = IS_ENABLED(CONFIG_THINGSET_STORAGE_INHIBIT_OVERWRITE);
+
+void thingset_storage_save_queued(bool force)
+{
+    if (force) {
+        storage_save_inhibit = false;
+    }
+
+    thingset_sdk_reschedule_work(&storage_work, K_NO_WAIT);
+}
+
+static void thingset_storage_update_handler()
+{
+    thingset_storage_save_queued(!storage_save_inhibit);
+}
+
+static void thingset_storage_save_handler(struct k_work *work)
+{
+    struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+
+    if (!storage_save_inhibit) {
+        thingset_storage_save();
+    }
```

Comment:
I think this probably wants something like:

```
} else {
    LOG_WRN("Not saving because previous load failed.");
}
```

...otherwise it is just silently failing to save changes.